### PR TITLE
Add docs for perma-failed deployments

### PR DESCRIPTION
**Pull request by kargakis**
@kubernetes/deployment @erictune @smarterclayton

---

This change is [Reviewable](https://reviewable.kubernetes.io/reviews/kubernetes/kubernetes.github.io/1337)

### CODE REVIEW

**Comment by kargakis** [comment]
Docs for https://github.com/kubernetes/kubernetes/pull/19343

**Comment by janetkuo** [comment]
Note that since perma-failed deployment isn't merged/released yet, this should either target 1.5 release branch (not yet created) or be added to 1.5 milestone (not yet created) and hold merging until 1.5.

File: docs/user-guide/deployments.md [hunk 1]
```diff
@@ -451,7 +451,79 @@ nginx-deployment-2035384211   0         0         1h
 nginx-deployment-3066724191   0         0         1h
 ```
 
-Note: A paused Deployment cannot be scaled at this moment, and we will add this feature in 1.3 release, see [issue #20853](https://github.com/kubernetes/kubernetes/issues/20853). You cannot rollback a paused Deployment either, and you should resume a Deployment first before doing a rollback. 
+Note: You cannot rollback a paused Deployment. You should resume it first before doing a rollback.
```
Comment (janetkuo) [inline docs/user-guide/deployments.md]: Could you create a separate PR for this? This change can go live now.

Comment (kargakis) [inline docs/user-guide/deployments.md]: Opened https://github.com/kubernetes/kubernetes.github.io/pull/1343

File: docs/user-guide/deployments.md [hunk 2]
```diff
@@ -451,7 +451,79 @@ nginx-deployment-2035384211   0         0         1h
...
+## Failed Deployment
+
+There will be times your Deployment will get stuck trying to deploy its newest Replica Set without ever completing
+(due to insufficient quota, readiness probe failures, image pull error, or any application runtime misconfiguration).
+In those cases you can specify a timeout parameter in the spec (`spec.progressDeadlineSeconds`) that denotes the
+seconds you want to wait for your Deployment to report any progress before the controller stops retrying to deploy
+it. Once a Deployment times out, a related timeout Condition will be added in the status of the Deployment. Note that
+if the error is transient eg. you have specified a low timeout or you don't have enough quota and all you need is to
+scale down your Deployment or scale down other Pods you may be running then once the deployment controller notices
+that your Deployment is complete, it will transition the failure condition into a succeeded one.
```
Comment (kargakis) [inline docs/user-guide/deployments.md]: done

**Comment by janetkuo** [comment]
Note: this depends on #19343 (merged) and #35691

**Comment by kargakis** [comment]
> Note: this depends on kubernetes/kubernetes#19343 (merged) and kubernetes/kubernetes#35691

Also https://github.com/kubernetes/kubernetes/pull/36171 for the cli changes. All three of them are now merged.

**Comment by janetkuo** [comment]
@kargakis not sure if you saw @jaredbhatti's mail regarding docs targeting 1.5 features. Would you open another PR for this change, targeting 1.5 branch (instead of master), and assign or cc me there?

**Comment by janetkuo** [comment]
Ref feature issue https://github.com/kubernetes/features/issues/122

**Comment by kargakis** [comment]
Opened https://github.com/kubernetes/kubernetes.github.io/pull/1699

**Pull request closed without merging**